src/clr-angular/data/datagrid/_datagrid.clarity.scss

```diff
@@ -915,9 +915,146 @@
   }
 
   /*
-        The following classes are used by the renderer when performing internal operations.
-        Using the browser calculations improves rendering performance.
-    */
+   * The following are styles for Detail Pane
+   */
+  .datagrid-outer-wrapper {
+    display: flex;
+    width: 100%;
+    overflow: hidden;
+  }
+
+  .datagrid-detail-open {
+    flex: 1 0 0;
```

Comment:
Playing with the demo I'm seeing the column re-size when the detail pane is open. 

Using `flex: none` here seems to fix it.